cli/src/commands/git/clone.rs

```diff
@@ -144,6 +144,12 @@ pub fn cmd_git_clone(
             .ok();
         }
     }
+    if args.colocate {
+        writeln!(
+            ui.warning_default(),
```

Comment:
nit: maybe `ui.hint_default()` is better? This only matters for people who casually run `git clean -f`, and we might want to add `hints.git-clean-colocated=false` to suppress it.

Reply:
I don't think we have any hints that (or at least very few) that you can opt out of so far. I can leave this PR up for a while so we can hear from others if they want a hint or a warning you can opt out from.

Reply:
I think @zx8 should requrest the flag on #5766, since that is the issue for disabling certain hints.

Reply:
I'll merge this now and then we have about a week to test it before next release. If people find it annoying, we can roll it back or add a config for it (and I would prefer to make it a warning if we have an opt-out anyway).